pkg/antctl/raw/multicluster/create/clusterset.go

```diff
@@ -0,0 +1,156 @@
+}
+
+func NewClusterSetCmd() *cobra.Command {
+	command := &cobra.Command{
+		Use:     "cluster-set",
+		Args:    cobra.MaximumNArgs(1),
+		Short:   "Create a ClusterSet in a Kubernetes cluster",
+		Long:    "Create a ClusterSet in a Kubernetes cluster",
+		Example: clusterSetExamples,
+		RunE:    clusterSetRunE,
+	}
+
+	o := &clusterSetOptions{}
+	clusterSetOpt = o
+	command.Flags().StringVarP(&o.namespace, "namespace", "n", "", "Namespace of the ClusterSet")
+	command.Flags().StringVarP(&o.leaderCluster, "leader-cluster", "", "", "Leader cluster of the ClusterSet")
+	command.Flags().StringVarP(&o.leaderClusterServer, "leader-server", "", "", "Leader cluster server address of the ClusterSet")
+	command.Flags().StringVarP(&o.leaderClusterPort, "leader-port", "", "6443", "Leader cluster port. If not set, use 6443 as default")
+	command.Flags().StringVarP(&o.leaderClusterNamespace, "leader-namespace", "", "", "Leader cluster namespace. If not set, use default namespace")
+	command.Flags().StringSliceVarP(&o.memberCluster, "member-cluster", "", nil, "Member clusters in the ClusterSet")
+	command.Flags().StringSliceVarP(&o.serviceAccount, "service-account", "", nil, "ServiceAccounts of the member clusters")
+	command.Flags().StringVarP(&o.secret, "secret", "", "", "Secret to access the leader cluster")
+
+	return command
+}
+
+func clusterSetRunE(cmd *cobra.Command, args []string) error {
+	if err := clusterSetOpt.validateAndComplete(); err != nil {
+		return err
+	}
+	if len(args) != 1 {
+		return fmt.Errorf("exactly one NAME is required, got %d", len(args))
+	}
+
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	restconfigTmpl := rest.CopyConfig(kubeconfig)
+	raw.SetupKubeconfig(restconfigTmpl)
+
+	scheme := k8sruntime.NewScheme()
+	if err = mcsscheme.AddToScheme(scheme); err != nil {
+		return err
+	}
+	if err = antreamcscheme.AddToScheme(scheme); err != nil {
+		return err
+	}
+	if err = k8sscheme.AddToScheme(scheme); err != nil {
+		return err
+	}
+	k8sClient, err := client.New(kubeconfig, client.Options{Scheme: scheme})
+	if err != nil {
+		return err
+	}
+	clusterSetName := args[0]
+	clusterSet := &multiclusterv1alpha1.ClusterSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterSetName,
+			Namespace: clusterSetOpt.namespace,
+		},
+		Spec: multiclusterv1alpha1.ClusterSetSpec{
+			Leaders: []multiclusterv1alpha1.MemberCluster{
+				{
+					ClusterID: clusterSetOpt.leaderCluster,
+					Secret:    clusterSetOpt.secret,
+					Server:    fmt.Sprintf("https://%s:%s", strings.Replace(clusterSetOpt.leaderClusterServer, "https://", "", 1), clusterSetOpt.leaderClusterPort),
+				},
+			},
+			Namespace: clusterSetOpt.leaderClusterNamespace,
+		},
+	}
+
+	for i := 0; i < len(clusterSetOpt.memberCluster); i++ {
+		clusterSet.Spec.Members = append(clusterSet.Spec.Members, multiclusterv1alpha1.MemberCluster{
+			ClusterID:      clusterSetOpt.memberCluster[i],
+			ServiceAccount: clusterSetOpt.serviceAccount[i],
```

Comment:
if member-cluster and service-account are slices, how can you guarantee that they are in right sequence and can pair correctly? better to use someway like `key=value` to set these pairs

Reply:
Maybe like this --clusters cluster-1=sa1, cluster2=sa2? 
Thanks for comment,  I'll do it.

Reply:
done